addons/project/models/project_task.py

```diff
@@ -1684,3 +1689,16 @@ def action_redirect_to_project_task_form(self):
             'url': '/web#model=project.task&id=%s&action=%s&view_type=form' % (self.id, self.env.ref('project.action_view_my_task').id),
             'target': 'new',
         }
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True): #TODO: multiple groupby ??
+        # A read_group can not be performed if records are grouped by personal_stage_type_id as it is a computed field.
+        # personal_stage_type_ids behaves like a M2O from the point of view of the user, we therefore use this field instead.
+        if 'personal_stage_type_id' in groupby and (not lazy or groupby[0] == 'personal_stage_type_id'):
+            groupby = ["personal_stage_type_ids" if field == "personal_stage_type_id" else field for field in groupby] # limitation: problem when both personal_stage_type_id and personal_stage_type_ids appear in read_group, but this has no functional utility
+            result = super().read_group(domain, fields, groupby, offset, limit, orderby, lazy)
+            for group in result:
+                group['personal_stage_type_id'] = group.pop('personal_stage_type_ids', False)
+                group['personal_stage_type_id_count'] = group.pop('personal_stage_type_ids_count', 0)
+            return result
```

Comment:
Yes I think so :+1: But I don't get the second part of your comment, why would it not be taken into account if not in first position?

Reply:
I mean indeed we manage multiple group by but when there is more then one group, we just assume personal_stage should be first but it could also be in any place. 🙂 

TBH, I am not against to not manage all the case since I think we don't manage it in the JS code. 🙂

Reply:
My bad 🙈